demo/android/STT/porcupine-stt-demo-app/src/main/java/ai/picovoice/porcupine/sttdemo/MainActivity.java

```diff
@@ -0,0 +1,320 @@
+                            public void run() {
+                                stateTextView.setText("Wake Word Detected");
+                                try {
+                                    // needs to stop porcupine manager before speechRecognizer can start listening.
+                                    porcupineManager.stop();
+                                } catch (PorcupineException e) {
+                                    displayError("Failed to stop porcupine.");
+                                    return;
+                                }
+
+                                new CountDownTimer(1000, 1000) {
+                                    @Override
+                                    public void onTick(long millisUntilFinished) {
+                                        // do nothing
+                                    }
+
+                                    @Override
+                                    public void onFinish() {
+                                        stateTextView.setText("Running STT");
+                                        intentWrapper.setBackgroundColor(getResources().getColor(R.color.colorPrimaryDark));
+                                        speechRecognizer.startListening(speechRecognizerIntent);
+                                    }
+                                }.start();
+                            }
+                        });
+                    }
+                });
+    }
+
+    private void configureKeywordSpinner() {
+        Spinner spinner = findViewById(R.id.keyword_spinner);
+
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(
+                this,
+                R.array.keywords,
+                R.layout.keyword_spinner_item);
+        adapter.setDropDownViewResource(R.layout.keyword_spinner_item);
+        spinner.setAdapter(adapter);
+
+        final ToggleButton recordButton = findViewById(R.id.record_button);
+
+        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parentView, View selectedItemView, int position, long id) {
+                if (recordButton.isChecked()) {
+                    stopService();
+                    recordButton.toggle();
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parentView) {
+                // Do nothing.
+            }
+        });
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        intentWrapper = findViewById(R.id.intentWrapper);
+        intentTextView = findViewById(R.id.intentView);
+        stateTextView = findViewById(R.id.stateText);
+
+        stateTextView.setText("Stopped");
+
+        // on android 11, RecognitionService has to be specifically added to android manifest.
```

Comment:
If this demo has a minimum Android version, we should mention this in the README

Reply:
On android 11, you need to add recognition service specifically on AndroidManifest.xml to use it. Lower android devices work fine.